src/main/java/net/fabricmc/loom/task/MigrateMappingsTask.java

```diff
@@ -179,7 +193,7 @@ private static void migrateMappings(Project project, MinecraftMappedProvider min
 		 * Since we only use intermediary names (and not descriptors) to match, and intermediary names are unique,
 		 * this will migrate methods that have had their signature changed too.
 		 */
-		public MappingsJoiner(TinyTree sourceMappings, TinyTree targetMappings, String fromNamespace, String toNamespace) {
+		private MappingsJoiner(TinyTree sourceMappings, TinyTree targetMappings, String fromNamespace, String toNamespace) {
```

Comment:
Leave it package private so that java doesnt generate a synthetic bridge constuctor.

Reply:
Java does that? The more you know.

Reply:
What's the issue with the synthetic bridge ?